internal/engine/watchmanager.go

```diff
@@ -128,14 +129,43 @@ func (w *WatchManager) diff(ctx context.Context, st store.RStore) (setup []Watch
 	return setup, teardown
 }
 
-func dependenciesMatch(d1 []string, d2 []string) bool {
-	if len(d1) != len(d2) {
+func watchRulesMatch(w1, w2 WatchableTarget) bool {
```

Comment:
have you considered using go-cmp like we do in model?
https://github.com/google/go-cmp